src/Fantomas.Core.Tests/DotGetTests.fs

```diff
@@ -1378,3 +1378,34 @@ let ``dotget inside a quotation, 2154`` () =
             .AsyncGetIndicator(indicatorIdVal)
     @@>)
 """
+
+[<Test>]
+let ``dotget break in for loop, 1222`` () =
+    formatSourceString
+        false
+        """
+module Foo =
+
+    let CreateReadOnlyAccounts (watchWalletInfo: WatchWalletInfo): Async<unit> =
+        async {
+            for etherCurrency in Currency.GetAll().Where(fun currency -> currency.IsEtherBased ()) do
+                do! ValidateAddress etherCurrency watchWalletInfo.EtherPublicAddress
+        }
+"""
+        config
+    |> prepend newline
+    |> should
+        equal
+        """
+module Foo =
+
+    let CreateReadOnlyAccounts (watchWalletInfo: WatchWalletInfo) : Async<unit> =
+        async {
+            for etherCurrency in
+                Currency
+                    .GetAll()
+                    .Where(fun currency -> currency.IsEtherBased())
+                do
```

Comment:
I'm a bit on the fence about whether this is still a good idea?
I know this is the sample I came up with a year ago but it looks weird now.
The important thing is that the `inExpr` is indented on the next line but what to do with the `do` or `arrow`.

```fsharp
for etherCurrency in
    Currency
        .GetAll()
        .Where(fun currency -> currency.IsEtherBased()) 
do
    do! ValidateAddress etherCurrency watchWalletInfo.EtherPublicAddress
```

Aligning the `for` and the `do` keyword seems consistent with long if/then or long match/with expressions. But that is not valid. It is for the `->` it seems.

```fsharp
for etherCurrency in
    Currency
        .GetAll()
        .Where(fun currency -> currency.IsEtherBased()) 
->
    do! ValidateAddress etherCurrency watchWalletInfo.EtherPublicAddress
```

Fantomas currently already seems to resolve the original problem. ([sample](https://fsprojects.github.io/fantomas-tools/#/fantomas/preview?data=N4KABGBEDGD2AmBTSAuKAzWAnMiAuAFolgMICuWWiAdtAJ5gCW1AOli9RBOZTfWxy5CwAOgDi%2BAIIAbaQAoAlAM7CIIgOpEqc9GU7QKVWgwC0APjAHexkQEkAzgFFCxAEIBDe4niKFYeLDKQgEAhGAAau7SjPDueIiS8PBU9va4LqSGfAwA7nHQBOpR0vi21JgizloACmQARtHQicmIqcrKmDj4WjxG9Eys7CrcWcZBqqISeDLySkMToprEiDp6lqP95uvW9HZOGR5ePgp%2B5uMQoRFRMXEJSSlp3cS92WB5eAVFsqXlsJUZtQajCa91a9kgABpwFAvHg8MwAObgtAAbWhEFAqkgABI8HQAA7INCQZh4SHorg4gBuUTIRLAaIWAAYoQtIGUkNQ8ABlRgAL2QrImABYKRAALoUgC%2BQoxYqguIJ9JJXPJWOxNOkdNQDPlEAAjLLhJAALLuAAeABlmIhLTQEYQ1RN9QAmJnyyVCGUUzHGxWEnWQGjwADy6Gt1EQ3LxJSdQmptPpjImLqN8cc1FD4ZtceN0nQkA90qNvvj-uVdVgsGkuYgCa1Sb1YAAzGnKWUvFg8AAxZhRAByiBy0Ujta4eCwdKLXpL8pxeIDxMr1bH9e1qKbwrbde5%2BPc0EQrkQnUQ1XcWHcAFt8MQxxAJ1PVJ6uN6hKXKeXA8ua22142FgArNuUC7vuh7HtgtqwDkxDQJ4iBlFSsBwfCsDUHeYAPog04vrO6oLhWVY-nOGqJjqyaqAAbMBkCgQeR4ngAqvihJYHBXiIchcSMGhGHoFEXg4RAr5cO%2BdafkuRGrqRDbkU2ADsNF0eBJ4kNInj2CQaH2A%2B0B4NgfECdhT7Fj6JEEV%2BUm-jJ666gsAAcSl7vREFUCaiCXnUt7Afx0iCSZM5mfhSqWSu1marZFHCAAnE5YEMZBWnSLxPlGUJYAiXKwWLlA37SRF-7OiyTa0c5CToPEpCwJel7uBhWHpZlYBiQqFmSWFJEFXJCz6oaJXKQlVDch5wLVilTa%2Bf5wjPsJeF%2Bm1uVWZ1ZEbj1qb9WVkgVcQw2XqNyXocBDUBbhQXzSF7XEeqXWrc6rYbWBkhYLAejwAAIog0R7ZV9WTsZ02mW%2B5kXVApL5StdnOluJVmuatjoAAKkQ1DcgQ2B4OoMSOsB7onbNZ1lgtKpkuFENRUI%2BpATDFrw0jNCOH5UZo12mPwNjm64wDgVA9lypg6Tsm3ao%2BrUdTcPlIw5ohqxcTYI45r4g8PGHU2AGc0IM0ZXNhMg8T4OC5DwuKWLABKiBwFg8Cs%2BzCzCurXCa01LXzrr-PLQb5NcPqjmm%2Bb2DwP2ZCecQYa2PEl7gsB%2BqNdrH5E5eZDSPCI6IN22C1XCxAIyFAuRU2%2BqxSVZsW-AJqJ8nNpp1gGc-TRBTnvulUAPo5FjBCFnjWsE3HruqrnhWqG6NGw09F50CGWDWjp1vt8Bdsx934lE2711k02Lp9WyI%2BUO44%2BT4wOmB8HWCh%2BHkf5wvPPnTlkAJ0njAp1XNdZzn7t5wsLrrWyo%2B7xPU94GXe%2Bj905xFriVeuF49LEBbm3DuXNTpXx1jfFefobqG2EC6e6W8LSRAbK4Zg8BEQzwwvZe2EpAaiWBsgvub8B4YOhtg803Y9B6WVvgzMRDYFzzIWAR2scl69xJrQ7qKYqaMLerAPAUx5aKzBMrYhwE1aX0obzUKV1UFrw-qLLe5cH42lcMlaAABrVwkCjH4HsCGFGV5EBJSDirBYk1-oawoVla%2BhEOqrw9uvY2bJBzDn0fgGCNBs6Eg%2BugZgjBULUEkJmdyx9z6OLSp3J2VCPHqLLGgz2EAXQ%2B2-tEBE1BmG0GibyQpcQKCIARrADkNA8DcXGkkxmyi3FIPSfrd%2BKZC75MqtQbiVIoLUARPEryWBwmROiYkiYTiWnNTSWojpdChDNmKjo%2B%2BEZbRXjqLENSsB7CIn8SnQyzSUn8NarrPK-cRGqGbJvCYkAZHEEYNeLkUQADSiBED4lqVyMopj3C0FnhNZJ8D8aIJ7jfS5wihbCGbF-e5BiAVGI2fYJ6L1MyDh0t4QBFdIwyKVtpX6j5QVd3BQIyFS0vGdJuVghF55BqIDegfaAWBnl9niPARi1BlYfWgOpC80TjlTRcdzFR7iFlXJhcshh9zHmspefU6Q0ZnpkB0pOfE0Y6CxlSicklqTVHEhQZkzRExmxiPuR8r5sMj6jLDIi6gyKbRTOFkyHhfDF7nIpZ4jR3iFjNm0fc5VwIAEIEFMC3VIqXzQnFEKEk9huz7J1DMqUQA))

I'm not sure any code change is really necessary. This PR can still serve as a regression test and close the issue.